components/match2/wikis/starcraft/match_legacy.lua

```diff
@@ -67,6 +67,7 @@ function MatchLegacy._storeGames(match, match2, options)
 			for key, item in pairs(playerdata) do
 				local k = mw.text.split(key or '', '_')
 				local l = tonumber(k[2])
+				---@diagnostic disable-next-line: cast-local-type
 				k = tonumber(k[1])
```

Comment:
Why is to number needed when it's immediately cast back into a string on the next line

Reply:
1) it is used as an index 1 line further down
2) even if i were to remove the tonumber it throws a warning since k is an array at that point